api_server.py

```
@@ -160,9 +162,12 @@ def check_rules(available_ids: List[int]) -> Dict:
         }
     with get_db() as conn:
         placeholders = ",".join("?" * len(available_ids))
+        # Use dynamic placeholders for IN clause with security suppression
+        query = f"SELECT role, skills FROM crew WHERE id IN ({placeholders})"
```

Ruff (0.15.6)

[error] Possible SQL injection vector through string-based query construction

(S608)